tapchannel/aux_sweeper.go

```diff
@@ -314,7 +314,7 @@ func (a *AuxSweeper) signSweepVpackets(vPackets []*tappsbt.VPacket,
 		// signature.
 		signingKey, leafToSign := applySignDescToVIn(
 			signDesc, vIn, &a.cfg.ChainParams,
-			tapscriptDesc.scriptTree.TapTweak(),
+			tapscriptDesc.scriptTree.TapTweak(), 0, false,
```

Comment:
IIUC the `index` value here has no meaning since `tweakWithIndex` is set to `false`.
Wouldn't the Functional Options Pattern be more elegant for `applySignDescToVIn`? You would get something like...
```go
signingKey, leafToSign := applySignDescToVIn(
	signDesc, vIn, s.cfg.ChainParams, tapscriptRoot,
	WithIndexTweak(baseJob.HTLC.HtlcIndex)
)

```
when you *do* want to tweak, and...
```go
signingKey, leafToSign := applySignDescToVIn(
	signDesc, vIn, &a.cfg.ChainParams,
	tapscriptDesc.scriptTree.TapTweak()
)
```
when you don't.

Reply:
Removed this whole code as it turned out to be wrong anyway.

Reply:
IIUC you created the func `TweakedRevocationKeyRing` which tweaks the `RevocationKey` and then use that keyring in the local and remote sweeps, both for timeout and success. Since we are *in* the aux_sweeper anyway, we know for a fact we need to tweak it with the `index` so no `tweakWithIndex` flag is needed.

Have I got that right?

Reply:
Yes, we need to tweak it for every sweep of an HTLC output. Since we currently only implement timeout and success sweeps, the tweak will only affect the internal key of the control block (because we're executing a script path in both cases). Once we implement revocation, then we'll also need to tweak the signing key, as then we'll use the key path.
Not sure if that's what you meant here?

Reply:
Thanks for the clarification. Yes, that's what I meant.